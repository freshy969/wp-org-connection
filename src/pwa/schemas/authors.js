import { schema } from 'normalizr';

export const author = new schema.Entity(
  'author',
  {},
  {
    processStrategy(entity) {
      entity.mst = 'author';
      entity.type = 'author';
      return entity;
    },
  },
);
export const authors = new schema.Array(author);
