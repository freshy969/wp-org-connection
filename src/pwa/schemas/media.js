import { schema } from 'normalizr';

export const media = new schema.Entity('media');
export const mediaArray = new schema.Array(media);
