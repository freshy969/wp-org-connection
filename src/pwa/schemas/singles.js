import { schema } from 'normalizr';

const author = new schema.Entity('author');
const media = new schema.Entity('media');
const taxonomy = new schema.Entity('taxonomy');
const taxonomies = new schema.Array(new schema.Array(taxonomy));
export const single = new schema.Entity('single')
single.define({
  _embedded: {
    author: [author],
    'wp:featuredmedia': [media],
    'wp:term': taxonomies,
    up: single,
  },
})
export const singles = new schema.Array(single);
