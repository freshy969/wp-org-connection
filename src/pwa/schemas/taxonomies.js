import { schema } from 'normalizr';

export const taxonomy = new schema.Entity('taxonomy');
export const taxonomies = new schema.Array(taxonomy);
