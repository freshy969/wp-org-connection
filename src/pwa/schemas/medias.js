import { schema } from 'normalizr';
import { author } from './authors';

export const media = new schema.Entity(
  'media',
  {
    _embedded: {
      author: [author],
    },
  },
  {
    processStrategy(entity) {
      entity.mst = 'media';
      return entity;
    },
  },
);

export const medias = new schema.Array(media);
