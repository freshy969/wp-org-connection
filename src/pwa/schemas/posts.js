/* eslint-disable no-underscore-dangle */
import { schema } from 'normalizr';
import { taxonomy } from './taxonomies';
import { author } from './authors';

const media = new schema.Entity('media');
const taxonomies = new schema.Array(new schema.Array(taxonomy));
export const post = new schema.Entity(
  'post',
  {},
  {
    processStrategy(entity) {
      // Get all taxonomies and generate a map so we can now later on which props are actually
      // taxonomies.
      if (entity._embedded && entity._embedded['wp:term']) {
        const taxonomiesMap = {};
        entity._embedded['wp:term'].forEach(term =>
          term.forEach(item => {
            const type = item.taxonomy === 'post_tag' ? 'tag' : item.taxonomy;
            taxonomiesMap[type] = taxonomiesMap[type] || [];
            taxonomiesMap[type].push(item.id);
          }),
        );
        return { ...entity, taxonomiesMap };
      }
      return entity;
    },
  },
);
post.define({
  _embedded: {
    author: [author],
    'wp:featuredmedia': [media],
    'wp:term': taxonomies,
    up: post,
  },
});
export const posts = new schema.Array(post);
