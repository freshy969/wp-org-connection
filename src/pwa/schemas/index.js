import { post, posts } from './posts';
import { taxonomy, taxonomies } from './taxonomies';
import { author, authors } from './authors';

export {
  post,
  posts,
  taxonomy,
  taxonomies,
  author,
  authors,
};
