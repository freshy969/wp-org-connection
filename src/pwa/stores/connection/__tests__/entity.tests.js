/* eslint-disable no-restricted-syntax */
import { autorun, observable } from 'mobx';
import { types, unprotect } from 'mobx-state-tree';
import { normalize } from 'normalizr';
import * as connect from '../';
import post60 from '../../../__tests__/post-60.json';
import page184 from '../../../__tests__/page-with-subpage.json';
import page211 from '../../../__tests__/page-211.json';
import { entity } from '../../../schemas';
import convert from '../../../converters';

const { entities } = normalize(post60, entity);

const Connection = types
  .model()
  .props(connect.props)
  .views(connect.views)
  .actions(connect.actions);

let connection = null;
beforeEach(() => {
  connection = Connection.create({});
  unprotect(connection);
});

describe('Connection â€º Entity', () => {
  test('Get entity shape when entity is not ready', () => {
    expect(connection.entity('someType', 'someId')).toMatchSnapshot();
  });

  test('Get single shape when entity is not ready', () => {
    expect(connection.entity('post', 60).ready).toBe(false);
    expect(connection.entity('post', 60).fetching).toBe(false);
    expect(connection.entity('post', 60).title).toBe('');
    expect(connection.entity('post', 60).link).toBe('/?p=60');
    expect(() => connection.entity('post', 60).pagedLink(3)).toThrow();
    expect(connection.entity('post', 60).taxonomy('category')).toEqual(observable([]));
    expect(connection.entity('post', 60).featured.ready).toBe(false);
    expect(connection.entity('post', 60).featured.sizes).toEqual(observable([]));
    expect(connection.entity('post', 60).author.name).toBe('');
  });

  test('Get taxonomy shape when entity is not ready', () => {
    expect(connection.entity('category', 3).ready).toBe(false);
    expect(connection.entity('category', 3).name).toBe('');
    expect(connection.entity('category', 3).link).toBe('/?cat=3');
    expect(connection.entity('category', 3).pagedLink(1)).toBe('/?cat=3');
    expect(connection.entity('category', 3).pagedLink(3)).toBe('/?cat=3&paged=3');
  });

  test('Get taxonomy shape when entity is not ready', () => {
    expect(connection.entity('tag', 10).ready).toBe(false);
    expect(connection.entity('tag', 10).name).toBe('');
    expect(connection.entity('tag', 10).link).toBe('/?tag_ID=10');
    expect(connection.entity('tag', 10).pagedLink(1)).toBe('/?tag_ID=10');
    expect(connection.entity('tag', 10).pagedLink(3)).toBe('/?tag_ID=10&paged=3');
  });

  test('Get author shape when entity is not ready', () => {
    expect(connection.entity('author', 4).ready).toBe(false);
    expect(connection.entity('author', 4).name).toBe('');
    expect(connection.entity('author', 4).avatar).toBe('');
    expect(connection.entity('author', 4).link).toBe('/?author=4');
    expect(connection.entity('author', 4).pagedLink(1)).toBe('/?author=4');
    expect(connection.entity('author', 4).pagedLink(2)).toBe('/?author=4&paged=2');
  });

  test('Get latest post taxonomy shape when entity is not ready', () => {
    expect(connection.entity('latest', 'post').ready).toBe(false);
    expect(connection.entity('latest', 'post').name).toBe('');
    expect(connection.entity('latest', 'post').link).toBe('/');
    expect(connection.entity('latest', 'post').pagedLink(1)).toBe('/');
    expect(connection.entity('latest', 'post').pagedLink(3)).toBe('/page/3');
  });

  test('Get latest movie taxonomy shape when entity is not ready', () => {
    expect(connection.entity('latest', 'movie').ready).toBe(false);
    expect(connection.entity('latest', 'movie').name).toBe('');
    expect(connection.entity('latest', 'movie').link).toBe('/');
    expect(connection.entity('latest', 'movie').pagedLink(1)).toBe('/');
    expect(connection.entity('latest', 'movie').pagedLink(3)).toBe('/page/3');
  });

  test('Get media shape when entity is not ready', () => {
    expect(connection.entity('media', 62).ready).toBe(false);
    expect(connection.entity('media', 62).link).toBe('/?attachement_id=62');
    expect(() => connection.entity('media', 62).pagedLink(2)).toThrow();
    expect(connection.entity('media', 62).author.name).toBe('');
    expect(connection.entity('media', 62).original.height).toBe(null);
    expect(connection.entity('media', 62).sizes).toEqual(observable([]));
  });

  test("Don't add an entity if it doesn't have type or id", () => {
    expect(connection.entity('media', 42).ready).toBe(false);
    connection.addEntity({ entity: { type: 'media' } });
    connection.addEntity({ entity: { id: 42 } });
  });

  test('Subscribe to ready before entity is ready', done => {
    expect(connection.entity('post', 60).ready).toBe(false);
    autorun(() => {
      if (connection.entity('post', 60).ready) done();
    });
    connection.addEntity({ entity: entities.single[60] });
  });

  test('Subscribe to link before entity is ready', done => {
    expect(connection.entity('post', 60).link).toBe('/?p=60');
    autorun(() => {
      if (
        connection.entity('post', 60).link === 'https://demo.worona.org/the-beauties-of-gullfoss/'
      )
        done();
    });
    connection.addEntity({ entity: entities.single[60] });
  });

  test('Subscribe to paged link before entity is ready', done => {
    expect(connection.entity('category', 3).pagedLink(2)).toBe('/?cat=3&paged=2');
    autorun(() => {
      if (
        connection.entity('category', 3).pagedLink(2) ===
          'https://demo.worona.org/wp-cat/photography/page/2' &&
        connection.entity('category', 3).pagedLink(1) ===
          'https://demo.worona.org/wp-cat/photography/'
      )
        done();
    });
    connection.addEntity({ entity: entities.taxonomy[3] });
  });

  test('Subscribe to single fields before entity is ready', done => {
    expect(connection.entity('post', 60).title).toBe('');
    autorun(() => {
      if (connection.entity('post', 60).title === 'The Beauties of Gullfoss') done();
    });
    connection.addEntity({ entity: entities.single[60] });
  });

  test('Fetching should be false after item is added', () => {
    expect(connection.entity('post', 60).fetching).toBe(false);
    connection.fetchingEntity({ type: 'post', id: 60 });
    expect(connection.entity('post', 60).fetching).toBe(true);
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).fetching).toBe(false);
  });

  test('Subscribe to fetching from false to true', done => {
    expect(connection.entity('post', 60).fetching).toBe(false);
    autorun(() => {
      if (connection.entity('post', 60).fetching) done();
    });
    connection.fetchingEntity({ type: 'post', id: 60 });
    expect(connection.entity('post', 60).ready).toBe(false);
  });

  test('Subscribe to fetching from true to false', done => {
    connection.fetchingEntity({ type: 'post', id: 60 });
    expect(connection.entity('post', 60).fetching).toBe(true);
    autorun(() => {
      if (!connection.entity('post', 60).fetching) done();
    });
    connection.addEntity({ entity: entities.single[60] });
  });

  test('Shape should be the same after fetching is started', () => {
    const authorId1 = connection.entity('post', 60).author.id;
    connection.fetchingEntity({ type: 'post', id: 60 });
    const authorId2 = connection.entity('post', 60).author.id;
    expect(authorId1).toEqual(authorId2);
    const featuredOriginal1 = connection.entity('post', 60).featured.original.url;
    connection.fetchingEntity({ type: 'post', id: 60 });
    const featuredOriginal2 = connection.entity('post', 60).featured.original.url;
    expect(featuredOriginal1).toEqual(featuredOriginal2);
    const link1 = connection.entity('post', 60).link;
    connection.fetchingEntity({ type: 'post', id: 60 });
    const link2 = connection.entity('post', 60).link;
    expect(link1).toEqual(link2);
    const taxonomies1 = connection.entity('post', 60).taxonomy('category');
    connection.fetchingEntity({ type: 'post', id: 60 });
    const taxonomies2 = connection.entity('post', 60).taxonomy('category');
    expect(taxonomies1).toEqual(taxonomies2);
  });

  test('Get taxonomies inside post before taxonomies are ready', () => {
    expect(connection.entity('post', 60).taxonomy('category').length).toBe(0);
    expect(connection.entity('post', 60).taxonomy('tag').length).toBe(0);
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).taxonomy('category').length).toBe(2);
    expect(connection.entity('post', 60).taxonomy('category')[0].id).toBe(3);
    expect(connection.entity('post', 60).taxonomy('tag').length).toBe(4);
    expect(connection.entity('post', 60).taxonomy('tag')[1].id).toBe(9);
  });

  test('Subscribe to taxonomies array before entity is ready', done => {
    expect(connection.entity('post', 60).taxonomy('category').length).toBe(0);
    autorun(() => {
      if (connection.entity('post', 60).taxonomy('category').length === 2) done();
    });
    connection.addEntity({ entity: entities.single[60] });
  });

  test('Subscribe to taxonomies array fields before entity is ready', done => {
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).taxonomy('category').length).toBe(2);
    expect(connection.entity('post', 60).taxonomy('category')[0].id).toBe(3);
    expect(connection.entity('post', 60).taxonomy('category')[0].name).toBe('');
    autorun(() => {
      if (connection.entity('post', 60).taxonomy('category')[0].name === 'Photography') done();
    });
    connection.addEntity({ entity: entities.taxonomy[3] });
  });

  test('Get featured inside post before entity is ready', () => {
    expect(connection.entity('post', 60).featured.id).toBe(null);
    expect(connection.entity('post', 60).featured.sizes).toEqual(observable([]));
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).featured.id).toBe(62);
    expect(connection.entity('post', 60).featured.title).toBe('');
  });

  test('Subscribe to featured fields before entity is ready', done => {
    expect(connection.entity('post', 60).featured.original.width).toBe(null);
    autorun(() => {
      if (connection.entity('post', 60).featured.original.width === 5000) done();
    });
    connection.addEntity({ entity: entities.single[60] });
    connection.addEntity({ entity: entities.media[62] });
  });

  test('Get author inside post before entity is ready', () => {
    expect(connection.entity('post', 60).author.id).toBe(null);
    expect(connection.entity('post', 60).author.name).toBe('');
    expect(connection.entity('media', 62).author.id).toBe(null);
    expect(connection.entity('media', 62).author.name).toBe('');
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).author.id).toBe(4);
    expect(connection.entity('post', 60).author.name).toBe('');
    connection.addEntity({ entity: entities.media[62] });
    expect(connection.entity('media', 62).author.id).toBe(2);
    expect(connection.entity('media', 62).author.name).toBe('');
  });

  test('Subscribe to author fields before entity is ready', done => {
    expect(connection.entity('post', 60).author.name).toBe('');
    autorun(() => {
      if (connection.entity('post', 60).author.name === 'Alan Martin') done();
    });
    connection.addEntity({ entity: entities.single[60] });
    connection.addEntity({ entity: entities.author[4] });
  });

  test('Get meta inside post before entity is ready', () => {
    expect(connection.entity('post', 60).headMeta.title).toBe('');
    expect(connection.entity('category', 3).headMeta.title).toBe('');
    expect(connection.entity('media', 62).headMeta.title).toBe('');
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).headMeta.title).toBe(
      convert(entities.single[60]).headMeta.title,
    );
    connection.addEntity({ entity: entities.taxonomy[3] });
    expect(connection.entity('category', 3).headMeta.title).toBe(
      convert(entities.taxonomy[3]).headMeta.title,
    );
    connection.addEntity({ entity: entities.media[62] });
    expect(connection.entity('media', 62).headMeta.title).toBe('iceland - Demo Worona');
  });

  test('Subscribe to meta fields before entity is ready', done => {
    expect(connection.entity('post', 60).headMeta.title).toBe('');
    autorun(() => {
      if (connection.entity('post', 60).headMeta.title === 'The Beauties of Gullfoss - Demo Worona')
        done();
    });
    connection.addEntity({ entity: entities.single[60] });
  });

  test('Subscribe to media original before entity is ready', done => {
    expect(connection.entity('media', 62).original.height).toBe(null);
    autorun(() => {
      if (connection.entity('media', 62).original.height === 3025) done();
    });
    connection.addEntity({ entity: entities.media[62] });
  });

  test('Subscribe to media sizes before entity is ready', done => {
    expect(connection.entity('media', 62).sizes.length).toBe(0);
    autorun(() => {
      if (connection.entity('media', 62).sizes.length === 6) done();
    });
    connection.addEntity({ entity: entities.media[62] });
  });

  test('Subscribe to taxonomy fields before entity is ready', done => {
    expect(connection.entity('category', 3).name).toBe('');
    autorun(() => {
      if (connection.entity('category', 3).name === 'Photography') done();
    });
    connection.addEntity({ entity: entities.taxonomy[3] });
  });

  test('Subscribe to author fields before entity is ready', done => {
    expect(connection.entity('author', 4).avatar).toBe('');
    autorun(() => {
      if (
        connection.entity('author', 4).avatar ===
        'https://secure.gravatar.com/avatar/eebc7d67bdc645559fd6634e0335a42b'
      )
        done();
    });
    connection.addEntity({ entity: entities.author[4] });
  });

  test('Get missing media (featured) inside ready post', () => {
    connection.addEntity({ entity: entities.single[60] });
    expect(connection.entity('post', 60).featured.id).toBe(62);
    expect(connection.entity('media', 62).ready).toBe(false);
    expect(connection.entity('media', 62).sizes).toEqual(observable([]));
    expect(connection.entity('post', 60).featured.ready).toBe(false);
    expect(connection.entity('post', 60).featured.sizes).toEqual(observable([]));
  });

  test('Get parent page', done => {
    expect(connection.entity('page', 184).parent).toBe(null);
    const { entities: entitiesFromPage184 } = normalize(page184, entity);
    connection.addEntity({ entity: entitiesFromPage184.single[184] });
    expect(connection.entity('page', 184).parent.id).toBe(211);
    expect(connection.entity('page', 184).parent.title).toBe('');
    autorun(() => {
      if (connection.entity('page', 184).parent.title === 'Audio TEST') done();
    });
    const { entities: entitiesFromPage211 } = normalize(page211, entity);
    connection.addEntity({ entity: entitiesFromPage211.single[211] });
    expect(connection.entity('page', 184).parent.id).toBe(211);
    expect(connection.entity('page', 184).parent.title).toBe('Audio TEST');
  });

  test('Get latest taxonomy', () => {
    expect(connection.entity('latest', 'post').ready).toBe(false);
    expect(connection.entity('latest', 'post').link).toBe('/');
    connection.addEntity({ entity: entities.taxonomy.post });
    expect(connection.entity('latest', 'post').ready).toBe(true);
    expect(connection.entity('latest', 'post').link).toBe('https://demo.worona.org');
    expect(connection.entity('latest', 'post').pagedLink(1)).toBe('https://demo.worona.org');
    expect(connection.entity('latest', 'post').pagedLink(2)).toBe('https://demo.worona.org/page/2');
  });
});
