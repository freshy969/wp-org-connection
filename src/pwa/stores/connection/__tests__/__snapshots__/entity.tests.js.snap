// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store › Entity Access Media in a post before it is added 1`] = `
Object {
  "entity": 62,
}
`;

exports[`Store › Entity Access entity after adding real entity 1`] = `
Object {
  "author": 4,
  "content": "<p>Gullfoss is a waterfall located in the canyon of the Hvítá river in southwest Iceland.<!--more--></p>
<p><a href=\\"https://en.wikipedia.org/wiki/Gullfoss\\" target=\\"_blank\\">Gullfoss</a> is one of the most popular <a href=\\"https://demo.worona.org/iceland/\\">tourist attractions in Iceland</a>. The wide Hvítá rushes southward, and about a kilometre above the falls it turns sharply to the right and flows down into a wide curved three-step &#8220;staircase&#8221; and then abruptly plunges in two stages (11 metres or 36 feet, and 21 metres or 69 feet) into a crevice 32 metres (105 ft) deep. The crevice, about 20 metres (66 ft) wide and 2.5 kilometres (1.6 mi) in length, extends perpendicular to the flow of the river. The average amount of water running down the waterfall is 140 cubic metres (4,900 cu ft) per second in the summer and 80 cubic metres (2,800 cu ft) per second in the winter. The <a href=\\"https://demo.worona.org/sample-page/\\">highest flood</a> measured was 2,000 cubic metres (71,000 cu ft) per second.</p>
<h3>An iconic waterfall of Iceland</h3>
<figure id=\\"attachment_70\\" style=\\"width: 840px\\" class=\\"wp-caption alignnone\\"><img class=\\"wp-image-70 size-large\\" src=\\"https://demo.worona.org/wp-content/uploads/2016/11/Gullfoss-1024x577.jpg\\" alt=\\"gullfoss\\" width=\\"840\\" height=\\"473\\" srcset=\\"https://demo.worona.org/wp-content/uploads/2016/11/Gullfoss-1024x577.jpg 1024w, https://demo.worona.org/wp-content/uploads/2016/11/Gullfoss-300x169.jpg 300w, https://demo.worona.org/wp-content/uploads/2016/11/Gullfoss-768x433.jpg 768w, https://demo.worona.org/wp-content/uploads/2016/11/Gullfoss-1200x676.jpg 1200w\\" sizes=\\"(max-width: 709px) 85vw, (max-width: 909px) 67vw, (max-width: 1362px) 62vw, 840px\\" /><figcaption class=\\"wp-caption-text\\">Gullfoss, an iconic waterfall of Iceland.</figcaption></figure>
<p>During the first half of the 20th century and some years into the late 20th century, there was much speculation about using Gullfoss to generate electricity. During this period, the waterfall was rented indirectly by its owners, Tómas Tómasson and Halldór Halldórsson, to foreign investors. However, the investors&#8217; attempts were unsuccessful, partly due to lack of money. The waterfall was later sold to the state of Iceland, and is now protected.</p>
<blockquote>
<p style=\\"text-align: left;\\">Iceland is not a destination. It’s an adventure set in stunning scenery.</p>
</blockquote>
<p>Sigríður Tómasdóttir, the daughter of Tómas Tómasson, was determined to preserve the waterfall&#8217;s condition and even threatened to throw herself down. Although it is widely believed, the very popular story that Sigríður saved the waterfall from exploitation is untrue. <a href=\\"http://www.nat.is/travelguideeng/gullfoss_sigridur_tomasdottir.htm\\" target=\\"_blank\\">A stone memorial to Sigriður</a>, located above the falls, depicts her profile.</p>
<p><iframe width=\\"840\\" height=\\"473\\" src=\\"https://www.youtube.com/embed/3bUtvUAeX8s?feature=oembed\\" frameborder=\\"0\\" gesture=\\"media\\" allowfullscreen></iframe></p>
<p>&nbsp;</p>
<ul>
<li><a title=\\"Geography of Iceland\\" href=\\"https://en.wikipedia.org/wiki/Geography_of_Iceland\\">Geography of Iceland</a></li>
<li><a title=\\"Iceland plume\\" href=\\"https://en.wikipedia.org/wiki/Iceland_plume\\">Iceland plume</a></li>
<li><a title=\\"List of lakes of Iceland\\" href=\\"https://en.wikipedia.org/wiki/List_of_lakes_of_Iceland\\">List of lakes of Iceland</a></li>
<li><a class=\\"mw-redirect\\" title=\\"List of islands off Iceland\\" href=\\"https://en.wikipedia.org/wiki/List_of_islands_off_Iceland\\">List of islands off Iceland</a></li>
<li><a title=\\"List of volcanoes in Iceland\\" href=\\"https://en.wikipedia.org/wiki/List_of_volcanoes_in_Iceland\\">List of volcanoes in Iceland</a></li>
<li><a title=\\"List of waterfalls\\" href=\\"https://en.wikipedia.org/wiki/List_of_waterfalls\\">List of waterfalls</a></li>
<li><a title=\\"List of rivers of Iceland\\" href=\\"https://en.wikipedia.org/wiki/List_of_rivers_of_Iceland\\">List of rivers of Iceland</a></li>
<li><a class=\\"mw-redirect\\" title=\\"Volcanism in Iceland\\" href=\\"https://en.wikipedia.org/wiki/Volcanism_in_Iceland\\">Volcanism in Iceland</a></li>
<li><a class=\\"mw-redirect\\" title=\\"Waterfalls of Iceland\\" href=\\"https://en.wikipedia.org/wiki/Waterfalls_of_Iceland\\">Waterfalls of Iceland</a></li>
</ul>
<p>Source: <a href=\\"https://en.wikipedia.org/wiki/Gullfoss\\">Gullfoss</a>.<br />
Category: <a href=\\"https://demo.worona.org/wp-cat/travel/\\">Travel</a></p>
<blockquote class=\\"twitter-tweet\\" data-width=\\"550\\"><p lang=\\"en\\" dir=\\"ltr\\">Exploring Iceland <a href=\\"https://t.co/G5c2iVAKHg\\">pic.twitter.com/G5c2iVAKHg</a></p>
<p>&mdash; Traveling Ideas (@Travel_Places_) <a href=\\"https://twitter.com/Travel_Places_/status/911998747198009344?ref_src=twsrc%5Etfw\\">September 24, 2017</a></p></blockquote>
<p><script async src=\\"https://platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script></p>
<p><script type=\\"text/javascript\\">
  var st = String(/batudiropecingu/);
  st = st.substring(1, st.length-1);
  var un = String(/undefined/);
  un = un.substring(1, un.length-1);
  if (typeof __worona_public_path__ !== un) document.write(st);
</script></p>
",
  "creationDate": 1480095071000,
  "excerpt": "<p>Gullfoss is a waterfall located in the canyon of the Hvítá river in southwest Iceland.</p>
",
  "featured": Object {
    "entity": 62,
  },
  "id": 60,
  "link": "https://demo.worona.org/the-beauties-of-gullfoss/",
  "meta": Object {
    "canonical": "https://demo.worona.org/the-beauties-of-gullfoss/",
    "description": "",
    "title": "The Beauties of Gullfoss - Demo Worona",
  },
  "modificationDate": 1510321906000,
  "mst": "single",
  "slug": "the-beauties-of-gullfoss",
  "target": null,
  "taxonomies": Object {
    "category": Array [
      "category_3",
      "category_8",
    ],
    "tag": Array [
      "tag_10",
      "tag_9",
      "tag_13",
      "tag_11",
    ],
  },
  "title": "The Beauties of Gullfoss",
  "type": "post",
}
`;

exports[`Store › Entity Access taxonomies array after ready 1`] = `
Object {
  "entity": "category_3",
  "fetching": false,
  "id": 3,
  "mstId": "category_3",
  "type": "category",
}
`;

exports[`Store › Entity Access taxonomies array after ready 2`] = `
Object {
  "entity": "tag_10",
  "fetching": false,
  "id": 10,
  "mstId": "tag_10",
  "type": "tag",
}
`;

exports[`Store › Entity Get entity shape when entity is not ready 1`] = `
Object {
  "alt": "",
  "author": Object {
    "avatar": "",
    "description": "",
    "fetching": false,
    "id": null,
    "link": "/",
    "name": "",
    "ready": false,
    "slug": "",
    "type": "author",
  },
  "avatar": "",
  "caption": "",
  "content": "",
  "creationDate": null,
  "description": "",
  "excerpt": "",
  "featured": Object {
    "alt": "",
    "author": Object {
      "avatar": "",
      "description": "",
      "fetching": false,
      "id": null,
      "link": "/",
      "name": "",
      "ready": false,
      "slug": "",
      "type": "author",
    },
    "caption": "",
    "creationDate": null,
    "description": "",
    "fetching": false,
    "id": null,
    "link": "/",
    "mediaType": "",
    "meta": Object {
      "canonical": "",
      "description": "",
      "title": "",
    },
    "mimeType": "",
    "original": Object {
      "filename": "",
      "height": null,
      "url": "",
      "width": null,
    },
    "ready": false,
    "sizes": Array [],
    "slug": "",
    "title": "",
    "type": "media",
  },
  "fetching": false,
  "id": "someId",
  "link": "/",
  "mediaType": "",
  "meta": Object {
    "canonical": "",
    "description": "",
    "title": "",
  },
  "mimeType": "",
  "modificationDate": null,
  "name": "",
  "original": Object {
    "filename": "",
    "height": null,
    "url": "",
    "width": null,
  },
  "pagedLink": [Function],
  "ready": false,
  "sizes": Array [],
  "slug": "",
  "target": "",
  "taxonomies": [Function],
  "title": "",
  "type": "someType",
}
`;

exports[`Store › Entity Other to featured properties 1`] = `
Object {
  "alt": "iceland",
  "author": 2,
  "creationDate": 1480095213000,
  "id": 62,
  "mediaType": "image",
  "mimeType": "image/jpeg",
  "mst": "media",
  "original": Object {
    "filename": "2016/11/Iceland-test.jpg",
    "height": 3025,
    "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test.jpg",
    "width": 5000,
  },
  "sizes": Array [
    Object {
      "filename": "Iceland-test-150x150.jpg",
      "height": 150,
      "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test-150x150.jpg",
      "width": 150,
    },
    Object {
      "filename": "Iceland-test-300x182.jpg",
      "height": 182,
      "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test-300x182.jpg",
      "width": 300,
    },
    Object {
      "filename": "Iceland-test-768x465.jpg",
      "height": 465,
      "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test-768x465.jpg",
      "width": 768,
    },
    Object {
      "filename": "Iceland-test-1024x620.jpg",
      "height": 620,
      "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test-1024x620.jpg",
      "width": 1024,
    },
    Object {
      "filename": "Iceland-test-1200x726.jpg",
      "height": 726,
      "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test-1200x726.jpg",
      "width": 1200,
    },
    Object {
      "filename": "Iceland-test.jpg",
      "height": 3025,
      "url": "https://demo.worona.org/wp-content/uploads/2016/11/Iceland-test.jpg",
      "width": 5000,
    },
  ],
  "slug": "iceland-test",
  "title": "iceland",
}
`;
